### PR TITLE
streamingccl: parallelize initial splits

### PR DESCRIPTION
**Pull request by msbutler**
During large scale initial scans, sending the initial splits can take a significant amount of time. This patch parallelize the splits across several workers on the job coordinator. To parallelize splits, we attempt to provide each worker with a large contiguous set of spans to minimize workers from working the same key space.

Epic: none

Release note: none

### CODE REVIEW

**Comment by msbutler** [comment]
According to a recent[ failure ](https://github.com/cockroachdb/cockroach/issues/115413) of the `c2c/tpcc/warehouses=500/duration=10/cutover=0` test, the initial splits took 10 seconds. With this patch, the initial splits took 4 seconds on this roachtest.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go [hunk 1]
```diff
@@ -244,6 +244,39 @@ func startDistIngestion(
 	return err
 }
 
+type partitionedSpans struct {
+	spans []roachpb.Span
+}
+
+var _ sort.Interface = &partitionedSpans{}
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go]: aha! I now understand why that type exists. Replaced.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go [hunk 2]
```diff
@@ -314,8 +348,33 @@ func createInitialSplits(
 	if err != nil {
 		return err
 	}
-	for _, partition := range topology.Partitions {
-		for _, span := range partition.Spans {
+
+	grp := ctxgroup.WithContext(ctx)
+	sortedSpans := sortSpans(topology.Partitions)
+	// Set the number of split workers to the number of partitions (i.e. src node count).
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go]: oof, yeah probably, though its ironically harder to plumb. Not that I had to grab it from the planner.

**Review by stevendanna — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=stevendanna

**Pull request merged**